x-pack/plugins/snapshot_restore/public/app/sections/home/repository_list/repository_list.tsx

```diff
@@ -125,16 +125,30 @@ export const RepositoryList: React.FunctionComponent<Props> = ({
     );
   }
 
+  const onRepositoryDelete = (repositoriesDeleted: Array<Repository['name']>): void => {
+    if (currentRepository && repositoriesDeleted.includes(currentRepository)) {
+      closeRepositoryDetails();
+    }
+    if (repositoriesDeleted.length) {
+      reload();
+    }
+  };
+
   return (
     <Fragment>
       {currentRepository ? (
-        <RepositoryDetails repositoryName={currentRepository} onClose={closeRepositoryDetails} />
+        <RepositoryDetails
+          repositoryName={currentRepository}
+          onClose={closeRepositoryDetails}
+          onRepositoryDelete={onRepositoryDelete}
```

Comment:
Minor nit, but maybe `onRepositoryDeleted` (past tense) is clearer? This also complements the name of it's parameter, `repositoriesDeleted`.